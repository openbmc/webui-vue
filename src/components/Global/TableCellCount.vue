<template>
  <div class="mt-2">
    <p v-if="!filterActive">
      {{ $t('global.table.items', { count: totalNumberOfCells }) }}
    </p>
    <p v-else>
      {{
        $t('global.table.selectedItems', {
          count: totalNumberOfCells,
          filterCount: filteredItemsCount,
        })
      }}
    </p>
  </div>
</template>

<script>
export default {
  props: {
    filteredItemsCount: {
      type: Number,
      required: true,
    },
    totalNumberOfCells: {
      type: Number,
      required: true,
    },
  },
  computed: {
    filterActive() {
      return this.filteredItemsCount !== this.totalNumberOfCells;
    },
  },
};
</script>
