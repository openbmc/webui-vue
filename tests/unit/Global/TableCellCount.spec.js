import { mount, createLocalVue } from '@vue/test-utils';
import TableCellCount from '@/components/Global/TableCellCount';

const localVue = createLocalVue();

describe('TableCellCount.vue', () => {
  const wrapper = mount(TableCellCount, {
    localVue,
    propsData: {
      filteredItemsCount: 5,
      totalNumberOfCells: 100,
    },
    mocks: {
      $t: (key) => key,
    },
  });
  it('should exist', () => {
    expect(wrapper.exists()).toBe(true);
  });
  it('should render filtered and totalnumber of items', () => {
    expect(wrapper.text()).toContain('global.table.selectedItems');
  });
  it('should render only totalnumber of items', async () => {
    await wrapper.setProps({ filteredItemsCount: 5, totalNumberOfCells: 5 });
    expect(wrapper.text()).toContain('global.table.items');
  });
  it('should render correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
