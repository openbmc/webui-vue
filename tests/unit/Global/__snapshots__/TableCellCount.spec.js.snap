// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableCellCount.vue should render correctly 1`] = `
<div
  class="mt-2"
>
  <p>
    global.table.items
  </p>
</div>
`;
